monix-reactive/jvm/src/main/scala/monix/reactive/compress/package.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2014-2020 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.reactive
+
+import java.time.Instant
+import java.util.zip.Deflater
+
+import monix.reactive.compress.internal.operators.{DeflateOperator, GunzipOperator, GzipOperator, InflateOperator}
+
+import scala.collection.compat.immutable.ArraySeq
+
+// From https://github.com/typelevel/fs2/blob/main/core/jvm/src/main/scala/fs2/compression.scala
+package object compress {
+  private[compress] val zeroByte: Byte = 0
+  private[compress] val gzipMagicFirstByte: Byte = 0x1f.toByte
+  private[compress] val gzipMagicSecondByte: Byte = 0x8b.toByte
+
+  private[compress] object gzipFlag {
+    def apply(flags: Byte, flag: Byte): Boolean = (flags & flag) == flag
+
+    def apply(flags: Byte, flag: Int): Boolean = (flags & flag) == flag
+
+    def ftext(flags: Byte): Boolean = apply(flags, FTEXT)
+
+    def fhcrc(flags: Byte): Boolean = apply(flags, FHCRC)
+
+    def fextra(flags: Byte): Boolean = apply(flags, FEXTRA)
+
+    def fname(flags: Byte): Boolean = apply(flags, FNAME)
+
+    def fcomment(flags: Byte): Boolean = apply(flags, FCOMMENT)
+
+    def reserved5(flags: Byte): Boolean = apply(flags, RESERVED_BIT_5)
+
+    def reserved6(flags: Byte): Boolean = apply(flags, RESERVED_BIT_6)
+
+    def reserved7(flags: Byte): Boolean = apply(flags, RESERVED_BIT_7)
+
+    val FTEXT: Byte = 1
+    val FHCRC: Byte = 2
+    val FEXTRA: Byte = 4
+    val FNAME: Byte = 8
+    val FCOMMENT: Byte = 16
+    val RESERVED_BIT_5 = 32
+    val RESERVED_BIT_6 = 64
+    val RESERVED_BIT_7: Int = 128
+  }
+
+  private[compress] object gzipExtraFlag {
+    val DEFLATE_MAX_COMPRESSION_SLOWEST_ALGO: Byte = 2
+    val DEFLATE_FASTEST_ALGO: Byte = 4
+  }
+
+  private[compress] object gzipCompressionMethod {
+    val DEFLATE: Byte = Deflater.DEFLATED.toByte
+  }
+
+  implicit class RichByteObservable(source: Observable[Byte]) {
```

Comment:
Alternatively, we could use `Transformer` that was brought recently but it would be a precedence. :D

I feel like it could be a bit nicer and more discoverable (`source.transform(gzip)`) than syntax import.
I would also move the methods to the top of the file so users don't have to scroll through private fields if they explore the file

WDYT?

Reply:
If the `Transformer` is a new way of doing such things then we should go with it.

I don't know if it is nicer or more discoverable but certainly it would be easier to use it from java. 

I will take a look at `Transformer` and try to use it.

Reply:
I don't use it too frequently myself, but many people like it, it's the equivalent of `via` / `transduce` / `through` etc